src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,963 @@
+/*
+Copyright 2013-2018 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /**********************************************
+     * fluid.orator
+     *
+     * A component for self voicing a web page
+     **********************************************/
+
+    fluid.defaults("fluid.orator", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            controller: ".flc-orator-controller",
+            content: ".flc-orator-content"
+        },
+        model: {
+            enabled: true,
+            play: false
+        },
+        components: {
+            tts: {
+                type: "fluid.textToSpeech"
+            },
+            controller: {
+                type: "fluid.orator.controller",
+                options: {
+                    parentContainer: "{orator}.container",
+                    model: {
+                        playing: "{orator}.model.play",
+                        enabled: "{orator}.model.enabled"
+                    }
+                }
+            },
+            selectionReader: {
+                type: "fluid.orator.selectionReader",
+                container: "{that}.container",
+                options: {
+                    model: {
+                        enabled: "{orator}.model.enabled"
+                    }
+                }
+            },
+            domReader: {
+                type: "fluid.orator.domReader",
+                container: "{that}.dom.content",
+                options: {
+                    model: {
+                        tts: {
+                            enabled: "{orator}.model.enabled"
+                        }
+                    },
+                    listeners: {
+                        "utteranceOnEnd.domReaderStop": [{
```

Comment:
No need for this to be an array since any more than one member of a namespaced listener would be discarded